include/pybind11/pytypes.h

```diff
@@ -613,6 +615,31 @@ struct tuple_item {
         }
     }
 };
+
+struct dict_item {
+    using key_type = object;
+
+    static object get(handle obj, handle key) {
+#if PY_MAJOR_VERSION >= 3
+        if (PyObject *result = PyDict_GetItemWithError(obj.ptr(), key.ptr())) {
+            return reinterpret_borrow<object>(result);
+        } else {
+            if (!PyErr_Occurred())
+                if (PyObject* key_repr = PyObject_Repr(key.ptr()))
```

Comment:
Why take the `repr`? According to this, that's not done by Python?

```
>>> try:
...     {}[42]
... except Exception as e:
...     exc = e
... 
>>> exc
KeyError(42,)
>>> exc.args
(42,)
>>> exc.args[0]
42
```

So you can just `PyErr_SetObject(PyExc_KeyError, key)` and save a level of indentation, or not?

Reply:
My bad 🤦‍♂️ no need to call `repr` on the key...

Reply:
At least my review is worth something, then ;-)